content/millennium/dstu2/care-provision/care-plan.md

```diff
@@ -76,26 +76,27 @@ _Implementation Notes_
 
  Name         | Required?                              | Type          | Description
 --------------|----------------------------------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------
- `_id`        | This, or one of `patient` or `subject` | [`token`]     | The logical resource id associated with the resource.
- `date`       | N                                      | [`date`]      | Time period plan covers. Example: `date=ge2016-08` or `date=le2017-01-24T12:00:00.000Z`
- `patient`    | This, or one of `_id` or `subject`     | [`reference`] | Who care plan is for. Example: `patient=1316024`
- `subject`    | This, or one of `_id` or `patient`     | [`reference`] | Who care plan is for. Must represent a Patient resource. May use the `:Patient` modifier. Example: `subject=Patient/1316024` or `subject:Patient=1316024`
- [`_count`]   | N                                      | [`number`]    | Number of results per page.
- `category`   | N                                      | [`token`]     | The scope of care plan being searched for. Examples: `category=assess-plan`, `category=http://argonaut.hl7.org|careteam`, `category=http://hl7.org/fhir/care-team-category|longitudinal`, or `category=careteam`
- `context`    | N                                      | [`reference`] | The encounter id of the care plan. Example: `context=1138`
+ `_id`        | This, or one of `patient` or `subject`                                    | [`token`]     | The logical resource id associated with the resource.
+ `date`       | N                                                                         | [`date`]      | Time period plan covers. Example: `date=ge2016-08` or `date=le2017-01-24T12:00:00.000Z`
+ `patient`    | This, or one of `_id` or `subject`, unless category is set to `encounter` | [`reference`] | Who care plan is for. Example: `patient=1316024`
+ `subject`    | This, or one of `_id` or `patient`, unless category is set to `encounter` | [`reference`] | Who care plan is for. Must represent a Patient resource. May use the `:Patient` modifier. Example: `subject=Patient/1316024` or `subject:Patient=1316024`
+ [`_count`]   | N                                                                         | [`number`]    | Number of results per page.
+ `category`   | This if `patient` or `subject` is set                                     | [`token`]     | The scope of care plan being searched for. Examples: `category=assess-plan`, `category=http://argonaut.hl7.org|careteam`, `category=http://hl7.org/fhir/care-team-category|longitudinal`, or `category=careteam`
+ `context`    | This, if category is set to `encounter`                                   | [`reference`] | The encounter id of the care plan. Example: `context=1138`
```

Comment:
Should also mention that if patient/subject or id are not provided then this would be required.

Reading back through the above it looks like you're pivoting the required required fields around whether category is set to encounter or not. This feels weird to me. I don't think we should add qualifiers like that to the required column.

I feel like it should be:
id: This, or one of 'patient', 'subject', or 'context'.
patient: This, or one of '_id', 'subject', or 'context'.
subject: This, or one of '_id', 'patient, or 'context'.
category: This if 'patient', 'subject', or 'context' are set.
context: This, or one of '_id', 'subject', or 'patient.

And then elsewhere it should be clarified that category must flex based on which other parameters are given.